### PR TITLE
Add plotting module

### PR DESCRIPTION
**Pull request by richypitman**
Added a plotting module with functionality to request plots via the pyscal CLI. One plot is made for each system (oil-water/gas-water/oil-gas) and each SATNUM. For curves interpolated using SCALrecommendation, the interpolated curves are plotted. By default, requesting plots will create relative permeability plots. There is an option to request a capillary pressure plot. Figures are saved in the current working directory. Some basic tests have also been added.

### CODE REVIEW

File: pyscal/pyscalcli.py [hunk 1]
```diff
@@ -149,6 +150,31 @@ def get_parser() -> argparse.ArgumentParser:
...
+        action="store_true",
+        default=False,
+        help=("Make and save relative permeability figures."),
+    )
+    parser.add_argument(
+        "--pc",
+        action="store_true",
+        default=False,
+        help=("Make and save capillary pressure figures."),
+    )
```
Comment (alifbe) [inline pyscal/pyscalcli.py:166]: Will it be better if we make it more explicit that it's referring to plotting Pc figure? Something like `plotpc`

File: pyscal/pyscalcli.py [hunk 2]
```diff
@@ -149,6 +150,31 @@ def get_parser() -> argparse.ArgumentParser:
...
+    parser.add_argument(
+        "--semilog",
+        action="store_true",
+        default=False,
+        help=(
+            "Plot relative permeability figures with log y-axis."
+            "Run both with and without this flag to plot"
+            "both linear and semi-log relperm plots."
+        ),
+    )
```
Comment (alifbe) [inline pyscal/pyscalcli.py:176]: Similar thing, should we have a more explicit name?

**Review by alifbe — CHANGES_REQUESTED** [review]
Could you please update the PR? Thank's a lot @richypitman

File: pyscal/pyscalcli.py (hunk 2 above)
Comment (richypitman) [reply pyscal/pyscalcli.py]: Also done in  a42fc88

File: tests/test_plotting.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        plotting.plotter(dummy_pyscal_list)
+
+
+def test_pyscal_list_attr():
+    # Check that the PyscalList class has an pyscal_list instance variable.
+    # This is access by the plotting module to loop through models to plot.
+    assert (
+        hasattr(PyscalList(), "pyscal_list") is True
+    ), "The PyscalList object should have a pyscal_list instance variable.\
+        This is accessed by the plotting module."
```
[the remark this answers is not in the document]
Comment (richypitman) [reply tests/test_plotting.py:85]: I have added a number of tests for the `plotting` module in a42fc88

**Comment by richypitman** [comment]
@alifbe I think this is ready to be approved 😃

**Review by alifbe — APPROVED** [review]
Looks good to me and don't forget to squash 😄 

Please consider my comment about CLI testing.

**Pull request merged**